Multisig/UI/Settings/OwnerKeyManagement/LedgerOwnerKey/LedgerKeyAddedViewController.swift

```diff
@@ -193,12 +193,30 @@ class AddDelegateKeyController {
     }
 
     func sendToBackend(delegateAddress: Address, signature: Data, completion: @escaping (Result<Void, Error>) -> Void) {
-        clientGatewayService.asyncCreateDelegate(safe: nil,
-                                                 owner: ownerAddress,
-                                                 delegate: delegateAddress,
-                                                 signature: signature,
-                                                 label: "iOS Device Delegate") { result in
-            completion(result.map { _ in () })
+        // to synchronize multiple async processes, we use DispatchGroup
+        let group = DispatchGroup()
+
+        Chain.all.forEach { chain in
+            // trigger request
+            group.enter()
+            clientGatewayService.asyncCreateDelegate(safe: nil,
+                    owner: ownerAddress,
+                    delegate: delegateAddress,
+                    signature: signature,
+                    label: "iOS Device Delegate",
+                    chainId: chain.id!) { result in
+                group.leave()
+            }
+        }
+        // wait until all requests finish, then call completion
+        let createDelegateRequestTimeoutInSeconds = 60 // one minute
```

Comment:
The name and comment do not add any meaning why is it 60 and not, say 49. Let's add the comment here `We use 60 seconds because it's a URLRequest's default timeout and we expect all requests to finish before that`.

And later we need to adjust the UI to show activity indicator or a progress bar to the user with the progress of requests (in a separate ticket and pull request)

Reply:
Fixed